docs/roles/integrator/integrating.md

```diff
@@ -0,0 +1,372 @@
+---
+id: integrating
+title: Integrating with NEAR
+sidebar_label: Integrating with NEAR
+---
+
+## Overview
+
+Integrating with NEAR is an opportunity for wallets, exchanges and oracles to take advantage of NEAR's scalability and performance for the benefit of their users.
+
+### Prerequisites
+
+- Hardware: Machine requirements are [here](https://docs-pr-136.onrender.com/docs/roles/validator/hardware)
+- Development:  Development environment requirements are available [here](https://docs.nearprotocol.com/docs/local-setup/local-dev-testnet)
+- Regulations: All regulations that apply to your jurisdiction are your sole responsibility including KYC / AML / CFT / BSA (Know Your Customer (KYC), Anti-Money Laundering (AML), Combating the Financing of Terrorism (CFT), Bank Secrecy Act (BSA))
+
+
+<blockquote class="info">
+<strong>did you know?</strong><br><br>
+
+The Near team recommends that custody wallets and exchanges track all shards to remove any possibility of invalid state transition. `tracking_shards` is a node configuration parameter available in the `~/.near/config.json` file which supports tracking all shards. You can also run 8 separate nodes that track each shard individually and route requests to the right nodes for higher performance.
+
+</blockquote>
+
+
+### Server Setup
+
+Please see [hardware requirements](https://docs-pr-136.onrender.com/docs/roles/validator/hardware) and details of how to [run a NEAR node](https://docs-pr-136.onrender.com/docs/local-setup/running-testnet).
+
+## Blockchain
+
+### Finality
+
+To learn whether a block is final or note, NEAR accepts a flag in calls to all RPC endpoints that indicates whether the requester wants the response as of the last block or the last *final* block.
+
+Exchanges should only use the last final block.
+
+<blockquote class="warning">
+<strong>work in progress</strong> <span>this feature is currently under development</span><br><br>
+
+Currently the list of parameters in the example below *does not include* the finality flag mentioned above.
+
+</blockquote>
+
+```sh
+# query format: account/<account_id>
+http post http://rpc.nearprotocol.com jsonrpc=2.0 method=query  \
+                                      params:="[\"account/test.near\",\"\"]" \
+                                      id="placeholder"
+# the request above returns view of account information
+# ------------------------------------------------------------------------------
+# response
+{
+  "id": "placeholder",
```

Comment:
if we show response, we somehow need to make sure that it will be up to date.
for example in Tatooine it's already extra `block_height` param added:
```
{
    "id": "123",
    "jsonrpc": "2.0",
    "result": {
        "amount": "100181737634864911400043",
        "block_height": 80110,
        "code_hash": "11111111111111111111111111111111",
        "locked": "10632884544298382540",
        "storage_paid_at": 69575,
        "storage_usage": 182
    }
}
```

Reply:
let me think about a way to setup some kind of versioning to let us know if examples have changes.  this was discussed with @eriktrautman and @vgrichina in a different context for `nearlib` examples here https://github.com/nearprotocol/docs/pull/112 but nothing conclusive yet.

Reply:
@ilblackdragon i added `block_height` to do docs but wasn't able to recreate using this query

```
http post http://rpc.tatooine.nearprotocol.com jsonrpc=2.0 method=query params:='["account/test",""]' id="placeholder"
```

getting `404`

i assumed i could just insert `tatooine` into the URL just like wallet and explorer.  what am i missing?

Reply:
`test` is visible as an account on tatooine via https://explorer.tatooine.nearprotocol.com/

Reply:
✅ done (edited sample to reflect this change)
⏳ still need clarity on how to verify this myself